Tests/ReactiveSwiftTests/SignalSpec.swift

```diff
@@ -213,6 +214,97 @@ class SignalSpec: QuickSpec {
 			}
 		}
 
+		describe("interruption") {
+			it("should not send events after sending an interrupted event") {
+				let queue: DispatchQueue
+				let counter = Atomic<Int>(0)
+
+				if #available(macOS 10.10, *) {
+					queue = DispatchQueue.global(qos: .userInitiated)
+				} else {
+					queue = DispatchQueue.global(priority: .high)
+				}
+
+				let (signal, observer) = Signal<Int, NoError>.pipe()
+
+				var hasSlept = false
+				var events = [Event<Int, NoError>]()
+
+				let sema = DispatchSemaphore(value: 0)
+
+				signal.observe { event in
+					if !hasSlept {
+						sema.signal()
+						// 100000 us = 0.1 s
+						usleep(100000)
+						hasSlept = true
+					}
+					events.append(event)
+				}
+
+				let group = DispatchGroup()
+
+				DispatchQueue.concurrentPerform(iterations: 10) { index in
+					queue.async(group: group) {
+						observer.send(value: index)
+					}
+
+					if index == 0 {
+						sema.wait()
+						observer.sendInterrupted()
+					}
+				}
+
+				group.wait()
+
+				expect(events.count) == 2
+				expect(events.count >= 2 ? events[1].isTerminating : false) == true
```

Comment:
This failure would be impossible to read. Can you separate the 2 checks with an `if`? That way it's possible to see what was checked and what the expectation was.